cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/CassandraBulkSourceRelation.java

```
@@ -270,16 +240,62 @@ private void persist(@NotNull JavaPairRDD<DecoratedKey, Object[]> sortedRDD, Str
         }
     }
 
+    private void waitForImportCompletion(TransportContext.CloudStorageTransportContext context,
+                                         long rowCount,
+                                         long totalBytesWritten,
+                                         boolean hasClusterTopologyChanged,
+                                         List<StreamResult> streamResults)
+    {
+        LOGGER.info("Waiting for Cassandra to complete import slices. rowCount={} totalBytes={} hasClusterTopologyChanged={}",
+                    rowCount,
+                    totalBytesWritten,
+                    hasClusterTopologyChanged);
+
+        List<StreamError> allErrors = streamResults.stream().flatMap(r -> r.failures.stream()).collect(Collectors.toList());
+        if (writerContext.job().isCoordinatedWriteEnabled())
+        {
+            if (!allErrors.isEmpty())
+            {
+                // log a warning and move on
```

Why move on in this case? If it's expected to be empty but isn't would that not be an error that should cause the job to fail early vs. continuing?

---

ok. I will throw illegal state to be strict.